servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ServiceTalkHttpAttributesGetter.java

```diff
@@ -109,20 +110,83 @@ private static List<String> getHeaderValues(final HttpHeaders headers, final Str
+
+        @Override
         @Nullable
+        public String getUrlFull(final HttpRequestMetaData request) {
+            HostAndPort effectiveHostAndPort = request.effectiveHostAndPort();
+            if (effectiveHostAndPort == null) {
+                return null;
+            }
+            StringBuilder sb = new StringBuilder();
+            String scheme = request.scheme();
+            sb.append(scheme == null ? "http" : scheme)
+              .append("://").append(effectiveHostAndPort.hostName());
+            if (effectiveHostAndPort.port() >= 0) {
+                sb.append(':').append(effectiveHostAndPort.port());
+            }
+            sb.append(request.requestTarget());
+            return sb.toString();
+        }
+
         @Override
-        public String getServerAddress(HttpRequestMetaData metaData) {
-            return null;
+        @Nullable
+        public String getServerAddress(final HttpRequestMetaData request) {
+            final HostAndPort effectiveHostAndPort = request.effectiveHostAndPort();
+            return effectiveHostAndPort != null ? effectiveHostAndPort.hostName() : null;
         }
 
         @Nullable
         @Override
         public Integer getServerPort(HttpRequestMetaData metaData) {
+            final HostAndPort effectiveHostAndPort = metaData.effectiveHostAndPort();
+            if (effectiveHostAndPort != null) {
+                return effectiveHostAndPort.port();
+            }
+            // No port. See if we can infer it from the scheme.
+            String scheme = metaData.scheme();
+            if (scheme != null) {
+                if ("http".equals(scheme)) {
+                    return 80;
+                }
+                if ("https".equals(scheme)) {
+                    return 443;
+                }
+            }
             return null;
         }
     }
 
     private static final class ServerGetter extends ServiceTalkHttpAttributesGetter
             implements HttpServerAttributesGetter<HttpRequestMetaData, HttpResponseMetaData> {
+
+        @Nullable
+        @Override
+        public String getClientAddress(HttpRequestMetaData metaData) {
+            return null;
+        }
+
+        @Nullable
+        @Override
+        public Integer getClientPort(HttpRequestMetaData metaData) {
+            return null;
+        }
+
+        @Override
+        public String getUrlScheme(final HttpRequestMetaData httpRequestMetaData) {
+            final String scheme = httpRequestMetaData.scheme();
+            return scheme == null ? "http" : scheme;
+        }
+
+        @Override
+        public String getUrlPath(final HttpRequestMetaData httpRequestMetaData) {
+            return httpRequestMetaData.path();
+        }
+
+        @Nullable
+        @Override
+        public String getUrlQuery(final HttpRequestMetaData httpRequestMetaData) {
+            return httpRequestMetaData.query();
```

Comment:
Query parameters and path are quite sensitive data. I can see how the absence of those may make tracing analysis more complicated, but I'm also worried about leaking critical info. Can we start with returning `null` and reconsider in a follow-up?

Reply:
Nvm, we will come back to this later.

Just another question: why these attributes moved to `ServerGetter` instead of being shared between client and server?

Reply:
This was a benign mistake in my previous update: `getUrlScheme`, `getUrlPath`, and `getUrlQuery` are only defined for the `HttpServerAttributesGetter` interface. For the client they did nothing.